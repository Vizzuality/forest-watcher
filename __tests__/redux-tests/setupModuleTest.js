import 'react-native';
import setupReducer, {
  initSetup,
  setSetupArea,
  setSetupCountry
} from 'redux-modules/setup';

describe('Redux Setup Module', () => {
  it('Initial reducer state', () => {
    expect(setupReducer(undefined, { type: 'NONE' })).toMatchSnapshot();
  });

  describe('Simple Actions: snapshot and reducer test', () => {
    // Snapshot tests the action object and the reducer state
    function simpleActionTest(action) {
      expect(action).toMatchSnapshot();
      expect(setupReducer(undefined, action)).toMatchSnapshot();
    }

    it('initSetup', () => {
      simpleActionTest(initSetup());
    });

    it('setSetupArea', () => {
      const mockCountryArea = {
        name: 'nameMock'
      };
      const mockCountryAreaFull = {
        name: 'nameMock', geojson: 'geojsonMock', wdpaid: 'wdpaidMock', id: 'idMock'
      };
      simpleActionTest(setSetupArea({ area: mockCountryArea, snapshot: 'snapshotMock' }));
      simpleActionTest(setSetupArea({ area: mockCountryAreaFull, snapshot: 'snapshotMock' }));
    });

    it('setSetupCountry', () => {
      const mockCountry = {
        name: 'nameMock', iso: 'isoMock', centroid: '{"centroidMock":"mockData"}', bbox: '{"bboxMock":"mockData"}'
      };
      simpleActionTest(setSetupCountry(mockCountry));
    });
  });
});
