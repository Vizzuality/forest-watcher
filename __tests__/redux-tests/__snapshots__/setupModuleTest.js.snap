// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Redux Setup Module Initial reducer state 1`] = `
Object {
  "area": Object {
    "geojson": null,
    "id": null,
    "name": "",
    "wdpaid": 0,
  },
  "country": null,
  "error": false,
  "snapshot": "",
}
`;

exports[`Redux Setup Module Simple Actions: snapshot and reducer test initSetup 1`] = `
Object {
  "type": "setup/INIT_SETUP",
}
`;

exports[`Redux Setup Module Simple Actions: snapshot and reducer test initSetup 2`] = `
Object {
  "area": Object {
    "geojson": null,
    "id": null,
    "name": "",
    "wdpaid": 0,
  },
  "country": null,
  "error": false,
  "snapshot": "",
}
`;

exports[`Redux Setup Module Simple Actions: snapshot and reducer test setSetupArea 1`] = `
Object {
  "payload": Object {
    "area": Object {
      "name": "nameMock",
    },
    "snapshot": "snapshotMock",
  },
  "type": "setup/SET_AOI",
}
`;

exports[`Redux Setup Module Simple Actions: snapshot and reducer test setSetupArea 2`] = `
Object {
  "area": Object {
    "name": "nameMock",
  },
  "country": null,
  "error": false,
  "snapshot": "snapshotMock",
}
`;

exports[`Redux Setup Module Simple Actions: snapshot and reducer test setSetupArea 3`] = `
Object {
  "payload": Object {
    "area": Object {
      "geojson": "geojsonMock",
      "id": "idMock",
      "name": "nameMock",
      "wdpaid": "wdpaidMock",
    },
    "snapshot": "snapshotMock",
  },
  "type": "setup/SET_AOI",
}
`;

exports[`Redux Setup Module Simple Actions: snapshot and reducer test setSetupArea 4`] = `
Object {
  "area": Object {
    "geojson": "geojsonMock",
    "id": "idMock",
    "name": "nameMock",
    "wdpaid": "wdpaidMock",
  },
  "country": null,
  "error": false,
  "snapshot": "snapshotMock",
}
`;

exports[`Redux Setup Module Simple Actions: snapshot and reducer test setSetupCountry 1`] = `
Object {
  "payload": Object {
    "bbox": "{\\"bboxMock\\":\\"mockData\\"}",
    "centroid": "{\\"centroidMock\\":\\"mockData\\"}",
    "iso": "isoMock",
    "name": "nameMock",
  },
  "type": "setup/SET_COUNTRY",
}
`;

exports[`Redux Setup Module Simple Actions: snapshot and reducer test setSetupCountry 2`] = `
Object {
  "area": Object {
    "geojson": null,
    "id": null,
    "name": "",
    "wdpaid": 0,
  },
  "country": Object {
    "bbox": Object {
      "bboxMock": "mockData",
    },
    "centroid": Object {
      "centroidMock": "mockData",
    },
    "iso": "isoMock",
    "name": "nameMock",
  },
  "error": false,
  "snapshot": "",
}
`;
